tools/api-proxy/secureapi-proxy.js

```diff
@@ -67,27 +149,30 @@ const handleRequest = async (request, env) => {
         status: response.status,
         headers: response.headers,
       });
+
     } catch (error) {
       // Handle any errors that might occur during the request forwarding
-      return new Response(`Error fetching data: ${error.message}`, { status: 500 });
+      return new Response('Error fetching data: ' + error.message, { status: 500 });
     }
-  } else {
+
+  }  else {
     // If not a POST or GET request, return 405 Method Not Allowed
     return new Response('Method Not Allowed', { status: 405 });
   }
-};
+}
 
-// eslint-disable-next-line no-unused-vars
 async function handleOptions(request) {
   // Handle CORS preflight requests.
   return new Response(null, {
     headers: {
-      'Access-Control-Allow-Origin': '*',
-      'Access-Control-Allow-Headers': '*',
+      "Access-Control-Allow-Origin" : "*",
```

Comment:
Can we set the policy to only allow requests from these domains:
icicidirect.com
main--icicidirect--aemsites.hlx.live
main--icicidirect--aemsites.hlx.page

Reply:
@sdmcraft yes regex is not supported thats true. 
>Let's do one thing, keep * here but programtically decline any request not coming from hlx.live , hlx.page, aem.live and aem.page.
yes that sounds good, will update the worker code accordingly

Reply:
@sdmcraft  updated here https://github.com/aemsites/icicidirect/pull/229/files#diff-1eb0952856b83fc76d69f34ae305ce32036479a81ce51c6c1853187d6613064bR77